test_regress/driver.py

```diff
@@ -2809,7 +2803,6 @@ def run_them() -> None:
     if 'TEST_REGRESS' in os.environ:
         sys.exit("%Error: TEST_REGRESS environment variable is already set")
     os.environ['TEST_REGRESS'] = os.getcwd()
-    os.environ['TERM'] = "dumb"
```

Comment:
Maybe just moving this into the `if` as you did is sufficient without other change?

Reply:
Unfortunately that's not enough. If you use `stdout=subprocess.PIPE` GDB will know it's output is no to a terminal and none of the features using terminal control sequences will work.